"keystore password was incorrect" when running TIM

Steps:

1. Build TIM from latest commit.
2. Start it as part of e.g. Training — or any other module — docker compose.

Starting TIM will fail with the following error:

```log
2025-01-13T09:50:12.480Z INFO [,] SecurityConfiguration: SecurityConfiguration filter:ee.eesti.authentication.configuration.CustomSessionAttributeSecurityFilter@4f114b
2025-01-13T09:50:12.662Z ERROR [,] SecurityConfiguration: Unable to initialize RSASSASigner 
java.io.IOException: keystore password was incorrect
```

This has broken in [this commit](https://github.com/buerokratt/TIM/commit/9a49c21309e6a94fcdd0d90e5521b2762a3dc6d2), previous commits work fine.

Running `./generate-keystore.sh` from the problematic commit gives `The operation couldn't be completed. Unable to locate a Java Runtime`. There are no instructions on this in the Readme so this is very confusing.

Some guide or at least a compatible Java version would be nice to have in the docs.

i ran into this bug when running `./generate-keystore.sh`

```
Generating 2,048 bit RSA key pair and self-signed certificate (SHA256withRSA) with a validity of 3,650 days
        for: CN=, OU=, O=, L=, ST=, C=
Generating 2,048 bit RSA key pair and self-signed certificate (SHA256withRSA) with a validity of 3,650 days
        for: CN=, OU=, O=, L=, ST=, C=
[INFO] Scanning for projects...
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-war-plugin/3.4.0/maven-war-plugin-3.4.0.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-war-plugin/3.4.0/maven-war-plugin-3.4.0.pom (8.4 kB at 17 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-enforcer-plugin/3.4.1/maven-enforcer-plugin-3.4.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-enforcer-plugin/3.4.1/maven-enforcer-plugin-3.4.1.pom (7.7 kB at 60 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/enforcer/enforcer/3.4.1/enforcer-3.4.1.pom
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/enforcer/enforcer/3.4.1/enforcer-3.4.1.pom (9.1 kB at 82 kB/s)
[INFO] 
[INFO] ----------------------------< rig.tim:tim >-----------------------------
[INFO] Building tim 0.0.1-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO] 
[INFO] >>> spring-boot-maven-plugin:3.2.5:run (default-cli) > test-compile @ tim >>>
[WARNING] The POM for ee.ria.commons:id-log:jar:1.0.0-SNAPSHOT is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.394 s
[INFO] Finished at: 2025-01-22T11:19:29+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project tim: Could not resolve dependencies for project rig.tim:tim:war:0.0.1-SNAPSHOT: Could not find artifact ee.ria.commons:id-log:jar:1.0.0-SNAPSHOT -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM eclipse-temurin:17-alpine AS build
+FROM eclipse-temurin:21-jdk-alpine AS build
 
 WORKDIR /workspace/app
 
@@ -12,7 +12,7 @@ COPY libs libs
 
 RUN ./mvnw install:install-file -Dfile=libs/id-log-${ID_LOG_VERSION}.jar \
     -DgroupId=ee.ria.commons -DartifactId=id-log -Dversion=${ID_LOG_VERSION} \
-    -Dpackaging=jar -DgeneratePom=true
+    -Dpackaging=jar -DgeneratePom=true && ./mvnw package -DskipTests=true
 
 COPY generate-keystore.sh /workspace/app/
 RUN chmod +x /workspace/app/generate-keystore.sh

